src/Schema/Table.php

```diff
@@ -861,6 +865,90 @@ private function _createIndex(
         return new Index($indexName, $columns, $isUnique, $isPrimary, $flags, $options);
     }
 
+    private function renameColumnInIndexes(string $oldName, string $newName): void
+    {
+        foreach ($this->_indexes as $key => $index) {
+            $modified = false;
+            $columns  = [];
+            foreach ($index->getColumns() as $columnName) {
+                if ($columnName === $oldName) {
+                    $columns[] = $newName;
+                    $modified  = true;
+                } else {
+                    $columns[] = $columnName;
+                }
+            }
+
+            if (! $modified) {
+                continue;
+            }
+
+            $this->_indexes[$key] = new Index(
+                $index->getName(),
+                $columns,
+                $index->isUnique(),
+                $index->isPrimary(),
+                $index->getFlags(),
+                $index->getOptions(),
+            );
```

Comment:
Should there be a wither for columns on `Index`, `ForeignKeyConstraint` and `UniqueConstraint` (marked internal if you don't think it should be exposed to end users)? This pattern is already present in the codebase.

Reply:
I want to have a builder (see https://github.com/doctrine/dbal/pull/6559). The problem with the wither is that the result of each modification will have to be valid, which is unnecessarily strict.